hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -469,6 +469,30 @@ public OzoneOutputStream createKey(String key, long size,
         .createKey(volumeName, name, key, size, replicationConfig, keyMetadata);
   }
 
+  /**
+   * Overwrite an existing key using optimistic locking. The existingKey must exist in Ozone to allow
+   * the new key to be created with the same name. Additionally, the existing Key must not have been
+   * modified since the time it's details were read. This is controlled by the updateID
+   * field in the existing Key. If the key is replaced or updated the updateID will change. If the
+   * updateID has changed since the existing Key was read, either the initial key create will fail,
+   * or the key will fail to commit after the data has been written as the checks are carried out
+   * both a key open and commit time.
+   *
+   * For now this feature only works on Object Store Buckets. FSO support will be added a later.
```

Comment:
```suggestion
   * This API allows to atomically update an existing key. The key read before invoking this API 
   * should remain unchanged for this key to be written. This is controlled by the updateID
   * field in the existing Key param. If the key is replaced or updated the updateID will change. If the
   * updateID has changed since the existing Key was read, either the initial key create will fail,
   * or the key will fail to commit after the data has been written as the checks are carried out
   * both a key open and commit time.
   *
   * For now this feature only works on Object Store Buckets. FSO support will be added a later.
```